runtimes/pythonrt/pythonrt.go

```diff
@@ -0,0 +1,453 @@
+
+	exports, err := entriesToValues(py.xid, entries)
+	if err != nil {
+		py.log.Error("can't create module entries", zap.Error(err))
+		return nil, fmt.Errorf("can't create module entries: %w", err)
+	}
+
+	killPy = false
+	py.run = ri
+	py.cbs = cbs
+	py.exports = exports
+	py.firstCall = true
+	py.dec = dec
+	py.enc = json.NewEncoder(conn)
+
+	return py, nil
+}
+
+func (py *pySVC) ID() sdktypes.RunID              { return py.xid.ToRunID() }
+func (py *pySVC) ExecutorID() sdktypes.ExecutorID { return py.xid }
+
+func (py *pySVC) Values() map[string]sdktypes.Value {
+	return py.exports
+}
+
+func (py *pySVC) Close() {
+	py.log.Info("closing")
+	// AK calls Close after `Run`, but we need the Python process running for `Call` as well.
+	// We kill the Python process once the initial `Call` is completed.
+}
+
+func (py *pySVC) initialCall(ctx context.Context, funcName string, payload []byte) (sdktypes.Value, error) {
+	defer func() {
+		py.log.Info("python done, killing")
+		if err := py.run.proc.Kill(); err != nil {
+			py.log.Warn("kill", zap.Int("pid", py.run.proc.Pid), zap.Error(err))
+		}
+		py.run.proc = nil
+	}()
+
+	// Initial run cal
+	msg := PyMessage{
+		Type:    "run",
+		Name:    funcName,
+		Payload: payload,
+	}
+	py.log.Info("initial call", zap.Any("message", msg))
+	if err := py.enc.Encode(msg); err != nil {
+		return sdktypes.InvalidValue, err
+	}
+
+	// Callbacks
+	for {
+		var msg PyMessage
+		py.log.Info("waiting for Python call")
+		if err := py.dec.Decode(&msg); err != nil {
+			py.log.Error("communication error", zap.Error(err))
+			return sdktypes.InvalidValue, err
+		}
+		py.log.Info("from python", zap.Any("message", msg))
+
+		if msg.Type == "done" {
+			break
+		}
+
+		// Generate activity, it'll call Python with the result
+		// The function name is irrelevant, all the information Python needs is in the Payload
+		fn, err := sdktypes.NewFunctionValue(py.xid, msg.Func.Name, msg.Payload, nil, pyModuleFunc)
+		if err != nil {
+			py.log.Error("create function", zap.Error(err))
+			return sdktypes.InvalidValue, err
+		}
+
+		py.log.Info("callback")
+		_, err = py.cbs.Call(
+			ctx,
+			py.xid.ToRunID(),
+			// The Python function to call is encoded in the payload
+			fn,
+			kittehs.Transform(msg.Func.Args, sdktypes.NewStringValue),
+			kittehs.TransformMap(msg.Func.Kw, func(key, val string) (string, sdktypes.Value) {
+				return key, sdktypes.NewStringValue(val)
+			}),
+		)
+		if err != nil {
+			py.log.Error("callback", zap.Error(err))
+			return sdktypes.InvalidValue, err
+		}
+	}
+
+	return sdktypes.Nothing, nil
+}
+
+func sliceToGo(values []sdktypes.Value) ([]any, error) {
+	out := make([]any, len(values))
+	for i, v := range values {
+		gv, err := valueToGo(v)
+		if err != nil {
+			return nil, err
+		}
+		out[i] = gv
+	}
+
+	return out, nil
+}
+
+// TODO: We just want the JSON of kwargs, this seems excessive. Ask Itay
+func valueToGo(v sdktypes.Value) (any, error) {
```

Comment:
i think you can just use `sdktypes.Unwrap` here.

Reply:
Doh! Thanks.